flink/v1.12/flink/src/main/java/org/apache/iceberg/flink/sink/ManifestOutputFileFactory.java

```diff
@@ -28,6 +28,7 @@
 import org.apache.iceberg.relocated.com.google.common.base.Strings;
 
 class ManifestOutputFileFactory {
+  private static final AtomicInteger fileCount = new AtomicInteger(0);
```

Comment:
I am not sure making it static is the right fix. Instead, we probably should fix the `generatePath` method and include the full table name in the path. 
```
  private String generatePath(long checkpointId) {
    return FileFormat.AVRO.addExtension(String.format("%s/%s-%05d-%d-%d-%05d", 
         fullTableName, flinkJobId, subTaskId,
        attemptNumber, checkpointId, fileCount.incrementAndGet()));
  }
```

Reply:
@stevenzwu I think that including the full table name can not solve this problem because the target table may be the same one.

Reply:
hmm. I didn't think about the case where target table can be the same. However, I also think the static variable may also not be enough. I think we can still get name collision when those committer operators run on different TMs. we may want to use this unique id from `StreamingRuntimeContext`

```
    /**
     * Returned value is guaranteed to be unique between operators within the same job and to be
     * stable and the same across job submissions.
     *
     * <p>This operation is currently only supported in Streaming (DataStream) contexts.
     *
     * @return String representation of the operator's unique id.
     */
    public String getOperatorUniqueID() {
        return operatorUniqueID;
    }
```

Reply:
@stevenzwu This is the operator unique id is fine.  I will fix it.

Reply:
@stevenzwu I have updated the pr. Could you please review it again? Thanks.